pkg/analysis/helpers/markers/analyzer.go

```diff
@@ -221,34 +228,98 @@ func extractMarker(comment *ast.Comment) Marker {
 		return Marker{}
 	}
 
-	id, expressions := extractMarkerIDAndExpressions(DefaultRegistry(), markerContent)
+	if isDeclarativeValidationMarker(markerContent) {
+		marker := extractDeclarativeValidationMarker(markerContent, comment)
+		if marker == nil {
+			return Marker{}
+		}
+
+		return *marker
+	}
+
+	return extractKubebuilderMarker(markerContent, comment)
+}
+
+func extractKubebuilderMarker(markerContent string, comment *ast.Comment) Marker {
+	id, arguments, payload := extractMarkerIDArgumentsAndPayload(DefaultRegistry(), markerContent)
 
 	return Marker{
-		Identifier:  id,
-		Expressions: expressions,
-		RawComment:  comment.Text,
-		Pos:         comment.Pos(),
-		End:         comment.End(),
+		Type:       MarkerTypeKubebuilder,
+		Identifier: id,
+		Arguments:  arguments,
+		Payload:    payload,
+		RawComment: comment.Text,
+		Pos:        comment.Pos(),
+		End:        comment.End(),
 	}
 }
 
-func extractMarkerIDAndExpressions(knownMarkers Registry, marker string) (string, map[string]string) {
+func extractMarkerIDArgumentsAndPayload(knownMarkers Registry, marker string) (string, map[string]string, *Payload) {
 	if id, ok := knownMarkers.Match(marker); ok {
-		return extractKnownMarkerIDAndExpressions(id, marker)
+		return extractKnownMarkerIDArgumentsAndPayload(id, marker)
+	}
+
+	return extractUnknownMarkerIDArgumentsAndPayload(marker)
+}
+
+func isDeclarativeValidationMarker(marker string) bool {
+	return strings.HasPrefix(marker, "k8s:")
+}
+
+func extractDeclarativeValidationMarker(marker string, comment *ast.Comment) *Marker {
+	tag, err := codetags.Parse(marker)
+	if err != nil {
+		return nil
 	}
 
-	return extractUnknownMarkerIDAndExpressions(marker)
+	return markerForTag(&tag, comment)
 }
 
-func extractKnownMarkerIDAndExpressions(id string, marker string) (string, map[string]string) {
-	return id, extractExpressions(strings.TrimPrefix(marker, id))
+func markerForTag(tag *codetags.Tag, comment *ast.Comment) *Marker {
```

Comment:
Does tag need to be a pointer?
```suggestion
func markerForTag(tag codetags.Tag, comment *ast.Comment) *Marker {
```